### DISCUSSION TITLE
How to configure automerge only for some packages?

### DISCUSSION THREAD

**Discussion by paololazzari** [renovatebot/renovate#29861] [Request Help]
### What would you like help with?

I would like help with my configuration

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, version 40.1.12

### Please tell us more about your question or problem

I have the following package rules:

```
"packageRules": [
    {
        "matchDatasources": ["docker"],
        "matchDepTypes": ["Kustomization"],
        "versioning": ["loose"]
    }
]
```

I have quite a few docker images, but I want to configure automerge only for some of them. How can I achieve this?

**Comment by RahulGautamSingh** [comment]
You can use `matchPackageNames` or `matchPackagePatterns` with `automerge`.

```json
"packageRules": [
    {
      "matchPackageNames": ["lodash"],
      "automerge": true
    }
  ]
```

**Reply by paololazzari** [reply]
But then what will happens to those packages that do not match? Will a PR still be opened?

**Reply by RahulGautamSingh** [reply]
Yes! we are only changing `automerge` behaviour here, it doesn't affect PR creation.

**Reply by HonkingGoose** [reply]
Please also read the [Renovate docs, key concepts, automerge](https://docs.renovatebot.com/key-concepts/automerge/), that explains a lot more about automerge.

Make sure you have proper tests _before_ you enable automerge! Or Renovate may automerge updates that break your build, or cause some other behavior problem.
